### PR TITLE
Optimize the bash script

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Optimzie the bash script when deploying.

Main changelog
1.  Use the `bin/hadoop` to get the hadoop deps.
2. Separate the log file for coordinator and server.

### Why are the changes needed?
To be compatible with the other hadoop distribution Including Apache Hadoop, like CDH.

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?
Manual test.

### CODE REVIEW

**Comment by zuston** [comment]
@jerqi Could u help review this? I found the current deploy script is not compatible with our internal Hadoop distribution.
This PR is to solve this.

File: conf/log4j.shuffle_server.properties [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+
+log4j.rootCategory=INFO, RollingAppender
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.Threshold=INFO
+log4j.appender.console.target=System.err
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
+log4j.appender.RollingAppender=org.apache.log4j.RollingFileAppender
+log4j.appender.RollingAppender.File=./logs/shuffle_server.log
```
[the remark this answers is not in the document]
Comment (zuston) [reply conf/log4j.shuffle_server.properties:25]: In my test cases, i make the coordinator and shuffle server deployed to one node. So i think it's better to separate the log file. 
Although the log file content looks same now, but separation will make log content change more easily.

[the remark this answers is not in the document]
Comment (jerqi) [reply conf/log4j.shuffle_server.properties:25]: First, We don't recommend to deploy coordinator and server in one node.
 If we deploy multiple servers in one node, do we need configurations `conf/log4j.shuffle_server.properties.1`,`conf/log4j.shuffle_server.properties.2`?
Second, Hadoop DataNode and Hadoop NodeManager can be deployed in one node, but they share one log4j.properties.

[the remark this answers is not in the document]
Comment (jerqi) [reply conf/log4j.shuffle_server.properties:25]: If there are many applications, the server will occupy all the network bandwidth and coordinator will be unstable.

File: bin/start-coordinator.sh [hunk 2]
```diff
@@ -32,8 +32,13 @@ cd $COORDINATOR_HOME
 source "${COORDINATOR_HOME}/bin/rss-env.sh"
 source "${COORDINATOR_HOME}/bin/utils.sh"
 
+if [ -z "$HADOOP_HOME" ]; then
+  echo "No env HADOOP_HOME."
+  exit 1
+fi
+
```
[the remark this answers is not in the document]
Comment (zuston) [reply bin/start-coordinator.sh:39]: JAVA_HOME has been set in rss-env.sh. 
I can't reproduce it.

**Review by jerqi — APPROVED** [review]
LGTM

**Pull request merged**